pkg/proxy/iptables/proxier_test.go

```diff
@@ -6151,6 +6151,11 @@ func TestSyncProxyRulesRepeated(t *testing.T) {
 	fp := NewFakeProxier(ipt)
 	metrics.RegisterMetrics()
```

Comment:
just  reset them once the test finish

```suggestion
	metrics.RegisterMetrics()
         defer Metrics.reset()
	defer legacyregistry.Reset()
```

Reply:
If this makes it possible to run metric test in parallel in general, I think this should be a "standard pattern".

Reply:
tests that use metrics can never run in parallel because store data in global variables

Reply:
This will work for running them sequentially multiple times, the test has to clear the state once it finish